splice_spark2/src/main/spark2.2/com/splicemachine/spark2/splicemachine/SplicemachineContext.scala

```diff
@@ -577,7 +577,7 @@ class SplicemachineContext(options: Map[String, String]) extends Serializable {
    * @param schema
    * @param schemaTableName table to delete from
    */
-  def delete(rdd: JavaRDD[Row], schema: StructType, schemaTableName: String): Unit = {
+  def delete(rdd: JavaRDD[Row], schema: StructType, schemaTableName: String): Unit = if( !rdd.isEmpty ) {
```

Comment:
This could force a computation of the RDD, ideally we'd want to make sure the RDD is only computed once. 
What's the error if the RDD is empty?

Reply:
The delete function (and it's the same for update) calls modifyOnKeys() which passes the rdd to sendData() which would put the data into Kafka, if there was data.  But in this case sendData() returns after having no data to pass, and then the sql is created and it hangs at 
`statement.executeUpdate(sql)`.  The code in the DB is probably polling Kafka when no data was put on the topic.  It looks like KafkaReadFunction would go into an infinite loop polling Kafka:
``` 
                    ConsumerRecords<Integer, Externalizable> records = null;
                    while (records == null || records.isEmpty()) {
                        records = consumer.poll( java.time.Duration.ofMillis(1000) );
                        if (TaskContext.get().isInterrupted()) {
                            consumer.close();
                            throw new TaskKilledException();
                        }
                    }
```